var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -241,6 +241,8 @@ class Openmpi(AutotoolsPackage):
         default=False,
         description='Do not remove mpirun/mpiexec when building with slurm'
     )
+    # Variants to use internal packages
+    variant('internal-hwloc', default=False, description='Use internal hwloc')
```

Comment:
Usually we try to avoid using vendored dependencies. Since Spack can't track them as node in the graph, it can't grant there won't be inconsistencies in the build. Is the situation somehow different here?

Reply:
With this option, Open MPI will not expose hwloc as a linkage dependency through its compiler wrapper (this is the core of the problem). In my mind, having an internal hwloc option is no different than having any other internal (private) library dependency within a library.

Reply:
@samuelkgutierrez is correct.   The Open MPI compiler scripts do not expose the internally built hwloc to the application.  I'm somewhat confused about why this variant should be an issue.  Its not the default, and helps solve the case where the end application or a dependency needs one version of hwloc, while the selected Open MPI version requires a different one.